### DISCUSSION TITLE
Pyswmm output module not working to get flood depths and pollutants from report file of SWMM

### DISCUSSION THREAD

**Discussion by MohammadTahirGhori** [pyswmm/pyswmm#325] [Q&A] [answered]
Hello all,
I am working on an optimization process through which I use pyswmm and run 1D model to get flood depths and pollutants mass. I use flood depths for flood damage as optimization function and also pollutants reduction as another optimization objective.
I need to get the flooding max depths at flooded nodes from report file of swmm and also the pollutants mass load from report file. 
I tried to use output model of pyswmm but it does not work. Can you help me to get flood depths and pollutants concentration (mass load) from report file of swmm after I run it. 

The error while I am trying to use output model from pyswmm.

from pyswmm import Output

ImportError: cannot import name 'Output' from 'pyswmm' (C:\Users\mgh004\Anaconda3\lib\site-packages\pyswmm\__init__.py)

**Comment by MohammadTahirGhori** [comment]
@jennwuu 
Can you please help me with this.

**Comment by jennwuu** [comment]
Hello!

Output module is only available with PySWMM version v1.1.1.

Please upgrade your PySWMM version.


Thanks!

**Reply by MohammadTahirGhori** [reply]
@jennwuu 
Hi 
Thanks for your reply. 
Actually before I was struggling with pollutants concentration to get from the simulation and then I upgraded the pyswmm version to v1.1.1 but unfortunately there was DLL error while my env was active and I am using conda. Then I reinstalled my python spyder. 
This time I am afraid if I upgrade pyswmm and get DLL error again.

**Comment by jennwuu** [comment]
What is your Python version? Is it regular python? It should be fine if it
is not conda.

What is your current PySWMM version?

**Reply by MohammadTahirGhori** [reply]
@jennwuu 
I am using python 3.9 with Spyder (anaconda3)
pyswmm version 0.6.2

**Comment by jennwuu** [comment, marked as answer]
You can try to reinstall PySWMM v1.1.1 again and check out these posts
about PATH issues in Anaconda.

https://stackoverflow.com/questions/54063285/numpy-is-already-installed-with-anaconda-but-i-get-an-importerror-dll-load-fail

Output module is only available for PySWMM version v1.1

**Reply by MohammadTahirGhori** [reply]
@jennwuu 
Thanks a lot. I will do it. 
Regards

**Reply by MohammadTahirGhori** [reply]
@jennwuu 
I tried all the methods of the link you provided. Adding different proposed changes to the system path. Unfortunately, the problem still exist.

**Reply by jennwuu** [reply]
@MohammadTahirGhori have you tried solution from this issue?

https://github.com/OpenWaterAnalytics/pyswmm/issues/298

**Reply by MohammadTahirGhori** [reply]
Hi @jennwuu 
This worked for me
(I also run into the same problem, since I install the pyswmm in anaconda env.
The solution works for me is to go to this directory below:
C:\Users\QM03JZ\Anaconda3\envs\pyswmm\Lib\site-packages\swmm\toolkit

and edit init.py by adding os.environ['CONDA_DLL_SEARCH_MODIFICATION_ENABLE']="1"

if platform.system() == "Windows":
libdir = os.path.join(os.path.dirname(file), "../../swmm_toolkit")

os.environ['CONDA_DLL_SEARCH_MODIFICATION_ENABLE']="1"
if hasattr(os, 'add_dll_directory'):    
    os.add_dll_directory(libdir)
else:
    os.environ["PATH"] = libdir + ";" + os.environ["PATH"])

Thanks a lot. 
Very helpful instructions.
Thanks for your time
